opentelemetry-sdk/src/opentelemetry/sdk/_metrics/__init__.py

```diff
@@ -46,95 +62,179 @@ def __init__(
 
-    def create_counter(self, name, unit=None, description=None) -> Counter:
-        # FIXME implement this method
-        pass
+    def create_counter(self, name, unit=None, description=None) -> APICounter:
+        # pylint: disable=protected-access
+        return self._meter_provider._create_counter(
+            self._instrumentation_info, name, unit, description
+        )
 
     def create_up_down_counter(
         self, name, unit=None, description=None
-    ) -> UpDownCounter:
-        # FIXME implement this method
-        pass
+    ) -> APIUpDownCounter:
+        # pylint: disable=protected-access
+        return self._meter_provider._create_up_down_counter(
+            self._instrumentation_info, name, unit, description
+        )
 
     def create_observable_counter(
         self, name, callback, unit=None, description=None
-    ) -> ObservableCounter:
-        # FIXME implement this method
-        pass
+    ) -> APIObservableCounter:
+        # pylint: disable=protected-access
+        return self._meter_provider._create_observable_counter(
+            self._instrumentation_info, name, unit, description
+        )
 
-    def create_histogram(self, name, unit=None, description=None) -> Histogram:
-        # FIXME implement this method
-        pass
+    def create_histogram(
+        self, name, unit=None, description=None
+    ) -> APIHistogram:
+        # pylint: disable=protected-access
+        return self._meter_provider._create_histogram(
+            self._instrumentation_info, name, unit, description
+        )
 
     def create_observable_gauge(
         self, name, callback, unit=None, description=None
-    ) -> ObservableGauge:
-        # FIXME implement this method
-        pass
+    ) -> APIObservableGauge:
+        # pylint: disable=protected-access
+        return self._meter_provider._create_observable_gauge(
+            self._instrumentation_info, name, unit, description
+        )
 
     def create_observable_up_down_counter(
         self, name, callback, unit=None, description=None
-    ) -> ObservableUpDownCounter:
-        # FIXME implement this method
-        pass
+    ) -> APIObservableUpDownCounter:
+        # pylint: disable=protected-access
+        return self._meter_provider._create_observable_up_down_counter(
+            self._instrumentation_info, name, unit, description
+        )
 
 
 class MeterProvider(APIMeterProvider):
     """See `opentelemetry._metrics.MeterProvider`."""
 
     def __init__(
         self,
+        metric_exporters: Sequence[MetricExporter] = (),
+        metric_readers: Sequence[MetricReader] = (),
+        views: Sequence[View] = (),
         resource: Resource = Resource.create({}),
         shutdown_on_exit: bool = True,
+        use_always_matching_view: bool = True,
     ):
-        self._resource = resource
+        self._lock = Lock()
         self._atexit_handler = None
+        self.__use_always_matching_view = use_always_matching_view
```

Comment:
what is this?

Reply:
From [this](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/sdk.md#view) part of the spec:

> If the Instrument could not match with any of the registered View(s), the SDK SHOULD provide a default behavior. The SDK SHOULD also provide a way for the user to turn off the default behavior via MeterProvider (which means the Instrument will be ignored when there is no match). Individual implementations can decide what the default behavior is, and how to turn the default behavior off.

So, if the user wants to turn off that behavior, then that attribute can be set to `False`.